File: 1161 Maximum Level Sum of a Binary Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def maxLevelSum(self, root: TreeNode) -> int:
        level, res, maxSum = 1, 1, root.val;
        q = collections.deque()
        q.append(root)
        
        while q:
            total = 0
            cnt = len(q)
            while cnt > 0:
                n = q.popleft()
                total += n.val
                if n.left: q.append(n.left)
                if n.right: q.append(n.right)
                cnt -= 1
                    
            if total > maxSum:
                maxSum, res = total, level            
            level += 1
        return res
        

File: 1161 Maximum Level Sum of a Binary Tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public int maxLevelSum(TreeNode root) {
        int level = 1, res = 1, maxSum = Integer.MIN_VALUE;
        Queue<TreeNode> q = new ArrayDeque<TreeNode>();
        q.offer(root);
        while (!q.isEmpty()) {            
            int sum = 0;            
            for (int cnt = q.size(); cnt > 0; --cnt) {
                TreeNode n = q.poll();
                sum += n.val;
                if (n.left != null) q.offer(n.left);
                if (n.right != null) q.offer(n.right);
            }
            
            if (sum > maxSum) {
                maxSum = sum;
                res = level;
            }
            
            ++level;
        }
        return res;
    }
}